lumicks/pylake/force_calibration/power_spectrum.py

```diff
@@ -156,3 +157,105 @@ def plot(self, **kwargs):
             plt.title(f"Blocked Power spectrum (N={self.num_points_per_block})")
         else:
             plt.title("Power spectrum")
+
+
+def identify_peaks(
+    power_spectrum: PowerSpectrum,
+    model_fun: callable,
+    return_indices: bool = False,
+    peak_cutoff: float = 20.0,
+    baseline: float = 1.0,
+):
+    """Identify peaks in the power spectrum, based on an exponential probability model with
+    rate parameter (lambda) = 1. This means that the power spectrum cannot be blocked or windowed.
+
+    Parameters
+    ----------
+    power_spectrum : PowerSpectrum
+        The power spectrum of the Brownian motion of the bead in a trap
+    model_fun : callable
+        A function of one argument, frequency, that gives the theoretical power spectrum. The
+        function is used to normalize the experimental power spectrum
+    return_indices: bool
+        Also return the array indices for the identified peaks (see below)
+    peak_cutoff: float
+        Indicates what value of the normalized spectrum is deemed abnormally high. Default is 20.0,
+        which corresponds to a chance of about 2 in 1E9 that a peak of that magnitude occurs
+        naturally in an exponential distribution with rate parameter = 1.0. The minimum is 15.0.
+    baseline: float
+        The baseline level a peak needs to drop down to. Lower means that exponentially more data is
+        considered to be part of the peak. The default is 1.0, and the range of baseline is [0.0,
+        peak_cutoff]. No fit or data smoothing is performed, the peak starts or ends at the first
+        data point that satisfies the criterion.
```

Comment:
Let's check if my probability skills are good enough: take an exponential distribution with rate parameter (lambda) = 1. The cumulative probability distribution is CDF(T) = 1 - exp(-T).  The chance of drawing a random value less than T is P(<T) = CDF(T), the chance of drawing a number higher than T is 1 - CDF(T).  Therefore, P(>T), the chance of drawing a value larger than T, is exp(-T).  

Drawing two numbers, each larger than T, would be P(>T) * P(>T) = exp(-T)^2. Three numbers is P(>T) * P(>T) * P(>T) = exp(-T)^3 etc. 

The level T, the base line, is used as a stopping criterion of the expansion of the peak. The probability of drawing N numbers above T is P(>T)^N. The chance that I will stop drawing numbers is therefore 1 - P(>T)^N. The code has the parameter `baseline`, which is T here. Using the default value of one, which is the rate parameter and thus the expectation value, I get the following probabilities:

```
N | P [%]
-----
1 | 63.2
2 | 86.5
3 | 95.0
4 | 98.2
5 | 99.3
```

Therefore, with the default value, there is a 99% chance that I stop expanding the range after 5 bins, if I would start outside a peak range. Practically, calibrations typically use at least 1 second worth of data. Therefore, 5 bins correspond to 5 Hz expansion, but typically less (because of probability and duration of the calibration). I think that those values are acceptable, even for low frequency bins. What do you think?

Reply:
Yep. Checks out!

The thing I was initially a bit concerned about was if the estimate is biased from having many spurious peaks; but on second thought, that actually doesn't matter much, since the exponential distribution is all tail. This will be pretty robust :)